Libplanet/Action/IAction.cs

```diff
@@ -1,16 +1,242 @@
+        /// Declares the set of <see cref="Address"/>es that an action requires
+        /// to fetch states.
+        /// <para>For example, if an action is someone healing someone it needs
+        /// to know a healer's remaning mana (i.e., MP) and a healing target's
+        /// remaning health (i.e., HP).  As a healing character belongs to
+        /// an account who signed the transaction and a healed character belongs
+        /// to an account who will receive the transaction, that healing
+        /// action's <see cref="RequestStates(Address, Address)"/> should
+        /// return a set of <paramref name="from"/> and <paramref name="to"/>.
+        /// </para>
+        /// </summary>
+        /// <param name="from">An <see cref="Address"/> of the account who
+        /// signed the transaction this action belongs to.</param>
+        /// <param name="to">An <see cref="Address"/> of the account who will
+        /// receive the transaction this action belongs to.</param>
+        /// <returns>An immutable set of <see cref="Address"/>es required by
+        /// this action.  Addresses are not limited to only <paramref
+        /// name="from"/> and <paramref name="to"/>.</returns>
+        /// <example>
+        /// The following example shows an action requiring states of
+        /// a transaction sender and a transaction receiver:
+        /// <code><![CDATA[
+        /// public IImmutableSet<Address> RequestStates(Address from,
+        ///                                             Address to)
+        /// {
+        ///     return new HashSet<Address> { from, to }.ToImmutableHashSet();
+        /// }
+        /// ]]></code>
+        /// </example>
+        IImmutableSet<Address> RequestStates(Address from, Address to);
 
+        /// <summary>
+        /// Executes the main game logic of an action.  This should be
+        /// <em>deterministic</em>.
+        /// <para>It is offered information such as a sender and a receiver of
+        /// the transaction this action belongs to, states immediately before
+        /// this action is executed, and a deterministic random seed, through
+        /// the <paramref name="context"/> object.
+        /// </para>
+        /// <para>Other &#x201c;bound&#x201d; information resides in the action
+        /// object in itself, as its properties (or fields).</para>
+        /// <para>A returned <see cref="AddressStateMap"/> object functions as
+        /// a delta which shifts from previous states to next states.</para>
+        /// </summary>
+        /// <param name="context">A context object which holds addresses that
+        /// signed and receives the transaction this action belongs to, states
+        /// immediately before this action is executed, and a PRNG object
+        /// which procudes deterministic random numbers.
+        /// See <see cref="IActionContext"/> for details.</param>
+        /// <returns>A map of <see cref="Address"/>es to update their states.
```

Comment:
A map of addresses with changed states. Only those addresses returned by this map will have their states updated globally.